contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,287 @@
+    constructor(
+        BaseStrategyConfig memory _baseConfig,
+        address _wethAddress,
+        address _ssvToken,
+        address _ssvNetwork,
+        address _feeAccumulator,
+        address _beaconChainDepositContract
+    )
+        InitializableAbstractStrategy(_baseConfig)
+        ValidatorAccountant(_wethAddress, _baseConfig.vaultAddress, _beaconChainDepositContract, _ssvNetwork)
+    {
+        SSV_TOKEN_ADDRESS = _ssvToken;
+        FEE_ACCUMULATOR_ADDRESS = _feeAccumulator;
+    }
+
+    /// @notice initialize function, to set up initial internal state
+    /// @param _rewardTokenAddresses Address of reward token for platform
+    /// @param _assets Addresses of initial supported assets
+    /// @param _pTokens Platform Token corresponding addresses
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+    }
+
+    /// @notice return the WETH balance on the contract that can be used to for beacon chain
+    /// staking - staking on the validators. Because WETH on this contract can be present as
+    /// a result of deposits and beacon chain rewards this function needs to return only WETH
+    /// that is present due to deposits.
+    function getWETHBalanceEligibleForStaking()
+        public
+        view
+        override
+        returns (uint256 _amount)
+    {
+        // if below amount results in a negative number there is a bug with accounting
+        _amount =
+            IWETH9(WETH_TOKEN_ADDRESS).balanceOf(address(this)) -
+            beaconChainRewardWETH;
+    }
+
+    /// @notice Collect accumulated WETH & SSV tokens and send to the Harvester.
+    function collectRewardTokens()
+        external
+        virtual
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // collect WETH from fee collector and wrap it into WETH
+        uint256 wethCollected = FeeAccumulator(FEE_ACCUMULATOR_ADDRESS)
+            .collect();
+
+        /* add up the WETH collected from the fee accumulator to beaconChainRewardWETH
+         * so it can be sent to the harvester in one swoop in the "_collectRewardTokens"
+         * step.
+         */
+        beaconChainRewardWETH += wethCollected;
+        _collectRewardTokens();
+    }
+
+    /// @dev Need to override this function since the strategy doesn't allow for all the WETH
+    /// to be collected. Some might be there as a result of deposit and is waiting for the Registrar
+    /// to be deposited to the validators.
+    function _collectRewardTokens() internal override {
+        uint256 rewardTokenCount = rewardTokenAddresses.length;
+        for (uint256 i = 0; i < rewardTokenCount; ++i) {
```

Comment:
yeah that is a good point this is done because of the SSV tokens. I was thinking that the Relayer would just send the SSV tokens to the strategy. Though given that SSV doesn't have a Chainlink oracle, we'd probably need to use Uniswap's TWAP.... we might skip some complexity and launch without the support for Harvester to swap the SSV for ETH. 

And add that functionality after the launch.

Reply:
Are we going to sell the SSV tokens? Given we need to pay the SSV Operators using SSV tokens, I was thinking the SSV rewards was more like a rebate. We can reuse the SSV rewards to pay for future SSV Operations.